### ISSUE TITLE
Missing dependencies

### ISSUE THREAD

**Issue by znck** [pnpm/pnpm#884]
### Code to reproduce the issue:

``` json
...
  "dependencies": {
    "bulma": "^0.5.1"
  },
  "devDependencies": {
    "@storybook/addon-actions": "^3.2.6",
    "@storybook/addon-links": "^3.2.6",
    "@storybook/vue": "^3.2.8",
    "ava": "^0.22.0",
    "babel-plugin-istanbul": "^4.1.4",
    "babel-plugin-transform-object-rest-spread": "^6.26.0",
    "babel-plugin-transform-runtime": "^6.23.0",
    "babel-preset-vue-app": "^1.2.1",
    "browser-env": "^3.2.0",
    "node-sass": "^4.5.3",
    "require-extension-hooks": "^0.3.0",
    "require-extension-hooks-babel": "^0.1.1",
    "require-extension-hooks-vue": "^0.4.0",
    "sass-loader": "^6.0.6",
    "sinon": "^3.2.1",
    "vue": "^2.4.2",
    "vue-template-compiler": "^2.4.2",
    "vue-test-utils": "git+https://github.com/vuejs/vue-test-utils.git#dev"
  },
...
```

### Expected behavior:
- `babel-preset-env` should be installed in `node_modules/babel-preset-vue-app/node_modules`.
- `devDependencies` should not appear in `dependencies`

### Actual behavior:
- No `node_modules` directory in `node_modules/babel-preset-vue-app`.
- `babel-preset-env` is not install at all (though available in `node_modules/.registry.npmjs.org`, but not linked).
- Post install, `devDependencies` are listed in `dependencies`

### Additional information:

<img width="552" alt="screen shot 2017-09-05 at 9 32 27 am" src="https://user-images.githubusercontent.com/2596484/30044855-c1882eb2-921d-11e7-8481-ccc3441c9e21.png">

 - `node -v` prints: v8.4.0
 - Windows, OS X, or Linux?: OS X

**Comment by zkochan** [comment]
> No node_modules directory in node_modules/babel-preset-vue-app.

`node_modules/babel-preset-vue-app` is just a symlink to `node_modules/.registry.npmjs.org/babel-preset-vue-app/1.3.0/node_modules/babel-preset-vue-app`. Node.js uses the real location to resolve requires. Dependencies of `babel-preset-vue-app` are in `node_modules/.registry.npmjs.org/babel-preset-vue-app/1.3.0/node_modules/`.

This weird structure works and is done to avoid circular symlinks.

You can read more about the node_modules layout that pnpm creates here: [Symlinked node_modules structure](https://github.com/pnpm/pnpm/blob/master/docs/symlinked-node-modules-structure.md)

**Comment by znck** [comment]
Should I add `readPackage` hook for `resolve`?

**Comment by zkochan** [comment]
if you have `resolve` in your deps, then you can try [this hook](https://github.com/pnpm/pnpm/issues/857#issuecomment-326746282)

**Comment by zkochan** [comment]
> devDependencies should not appear in dependencies

I fixed this in `pnpm@1.13.2`

**Comment by znck** [comment]
I confirm, devDependencies issue fixed.
